Doc/c-api/object.rst

```diff
@@ -575,3 +575,23 @@ Object Protocol
    has the :c:macro:`Py_TPFLAGS_MANAGED_DICT` flag set.
 
    .. versionadded:: 3.13
+
+.. c:function:: int PyUnstable_Object_EnableDeferredRefcount(PyObject *obj)
+
+   Enable `deferred reference counting <https://peps.python.org/pep-0703/#deferred-reference-counting>`_ on *obj*,
+   if supported by the runtime.  In the :term:`free-threaded <free threading>` build,
+   this allows the interpreter to avoid reference count adjustments to *obj*,
+   which may improve multi-threaded performance.  The tradeoff is
+   that *obj* will only be deallocated by the tracing garbage collector.
+
+   This function returns ``1`` if deferred reference counting was enabled on *obj*,
+   and ``0`` if deferred reference counting is not supported or if the hint was
+   ignored by the runtime. This function is thread-safe, and cannot fail.
+
+   This function is a no-op on builds with the :term:`GIL` enabled, which do
+   not support deferred reference counting. This also no-ops if *obj* is not
+   an object tracked by the garbage collector (see :func:`gc.is_tracked` and
+   :c:func:`PyObject_GC_IsTracked`).
+
+   .. versionadded:: next
```

Comment:
Maybe we want a note on the intended usage? It's not necessary now that the function can't fail, but perhaps it's still useful.
```suggestion
   This function is intended to be used soon after *obj* is created,
   by the code that creates it.

   .. versionadded:: next
```

Reply:
Hmm, maybe. That more applies to the internal `_PyObject_SetDeferredRefcount`, but this version is thread safe; users could apply it to any object they want (and they probably will, because it will be the only public way to ease reference count contention, as far as I know).